src/Sulu/Bundle/AdminBundle/Controller/AdminController.php

```diff
@@ -276,15 +270,23 @@ public function translationsAction(Request $request): Response
         return new JsonResponse($translations);
     }
 
-    public function metadataAction(string $type, string $key): Response
+    public function metadataAction(string $type, string $key, Request $request): Response
     {
         $user = $this->tokenStorage->getToken()->getUser();
 
+        $metadataOptions = $request->query->all();
+        $metadata = $this->metadataProviderRegistry->getMetadataProvider($type)->getMetadata($key, $user->getLocale(), $metadataOptions);
         $view = View::create(
-            $this->metadataProviderRegistry->getMetadataProvider($type)->getMetadata($key, $user->getLocale())
+            $metadata
         );
         $view->setFormat('json');
 
-        return $this->viewHandler->handle($view);
+        $response = $this->viewHandler->handle($view);
+
+        if (!$metadata->isCacheable()) {
+            $response->headers->addCacheControlDirective('no-store', !$metadata->isCacheable());
```

Comment:
Instead of reevaluating `!$metadata->isCacheable()` we could just pass true, since that will result in the same value always, right?